src/components/UserMenu/UserMenu.tsx

```diff
@@ -1,227 +1,103 @@
-        this.setState({ isClickable: value })
-      }
-    }, 250)
-  }
-
-  componentDidMount(): void {
-    this.mounted = true
-  }
-
-  componentWillUnmount(): void {
-    this.mounted = false
-  }
-
-  renderManaBalances = (): React.ReactNode => {
-    const { manaBalances, onClickBalance } = this.props
-
-    return (
-      <span className="dcl account-wrapper">
-        {Object.keys(manaBalances).map((network) => (
-          <Mana
-            key={network}
-            network={network as Network}
-            size="small"
-            className={onClickBalance ? 'clickable' : undefined}
-            title={`${manaBalances[network].toLocaleString()} MANA`}
-            href={config.get('ACCOUNT_URL')}
-          >
-            {Number(manaBalances[network].toFixed(2)).toLocaleString()}
-          </Mana>
-        ))}
-      </span>
-    )
-  }
-
-  render(): JSX.Element {
-    const {
-      avatar,
-      manaBalances,
-      isSignedIn,
-      isSigningIn,
-      isActivity,
-      hasActivity,
-      onSignOut,
-      onSignIn,
-      onClickProfile,
-      onClickActivity,
-      onClickSettings,
-      i18n,
-      menuItems
-    } = this.props
-
-    const { isOpen, isClickable } = this.state
-
-    const name = avatar ? avatar.name : null
-
-    const isSomeBalanceTooHigh = Object.values(manaBalances).some(
-      (balance) => Number(balance.toFixed(2)).toLocaleString().length > 5
-    )
+import './UserMenu.css'
 
-    return (
-      <Column align="right">
-        <Row className="dcl user-menu-wrapper">
-          <Menu.Item
-            className={isActivity ? 'activity-bell active' : 'activity-bell'}
+export const UserMenu = React.memo((props: UserMenuProps) => {
+  const {
+    isSignedIn,
+    isSigningIn,
+    manaBalances,
+    onClickSignIn,
+    onClickBalance,
+    onClickOpen,
+    onClickJumpIn,
+    onClickMenuItem,
+    ...signInProps
+  } = props
+
+  const [isOpen, setIsOpen] = useState(false)
+  const [trackingId, setTrackingId] = useState<string | null>(null)
+
+  const handleToggle = useCallback(
+    (event: React.MouseEvent<HTMLElement, MouseEvent>) => {
+      const trackId = uuidv4()
+      setIsOpen((prev) => {
+        if (!prev) {
+          setTrackingId(trackId)
+        }
+        if (!prev && onClickOpen) {
+          onClickOpen(event, trackId)
+        }
+        return !prev
+      })
+    },
+    [setIsOpen, onClickOpen]
+  )
+
+  const handleClose = useCallback(() => {
+    setIsOpen(false)
+  }, [setIsOpen])
+
+  const handleClickJumpIn = useCallback(
+    (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+      onClickMenuItem && onClickMenuItem(event, i18n.profile, trackingId)
+
+      setTimeout(
+        () => {
+          onClickJumpIn
+            ? onClickJumpIn(event)
+            : window.open(config.get('EXPLORER_URL'), '_blank', 'noopener')
+        },
+        onClickMenuItem ? 300 : 0
+      )
+    },
+    [onClickJumpIn, onClickMenuItem, trackingId]
```

Comment:
Profile seems to be missing from the dependencies here:
```suggestion
    [onClickJumpIn, onClickMenuItem, i18n.profile, trackingId]
```